### PR TITLE
Enable structured errors by default.

### PR DESCRIPTION
**Pull request by kenzieschmoll**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 1]
```diff
@@ -960,7 +960,16 @@ mixin WidgetInspectorService {
...
   /// to enhance the display of errors with rich formatting.
   bool isStructuredErrorsEnabled() {
-    return const bool.fromEnvironment('flutter.inspector.structuredErrors');
+    const String extension = 'flutter.inspector.structuredErrors';
+    bool enabled = false;
+    assert(() {
+      enabled = const bool.fromEnvironment(extension, defaultValue: true);
+      return true;
+    }());
+    if (enabled) {
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:969]: These 4 lines should be replaced by`return enabled;`

Comment (kenzieschmoll) [reply packages/flutter/lib/src/widgets/widget_inspector.dart:969]: if `const bool.fromEnvironment('flutter.inspector.structuredErrors')` is non-null though, we want to return it. If we don't return `const bool.fromEnvironment(extension);` at the end, we will automatically return false in profile mode, even though `bool.fromEnvironment(extension)` may have been true.

Comment (jacob314) [reply packages/flutter/lib/src/widgets/widget_inspector.dart:969]: This is really a debug mode only feature. Does it even function in profile mode? I would suggest just disabling it in profile mode and commenting that it is debug mode only. That is probably less surprising than intentionally having a different default value in profile and debug mode.

Comment (kenzieschmoll) [reply packages/flutter/lib/src/widgets/widget_inspector.dart]: good point. done.

**Pull request merged**